CONTRIBUTING.md

```diff
@@ -0,0 +1,196 @@
+* help wanted
+* doc incomplete
+* test improvement
+* any questions on project
+* and so on
+
+Also we must remind that when filling a new issue, please remember to remove the sensitive data from your post. Sensitive data could be password, secret key, network locations, private business data and so on.
+
+## Code and doc contribution
+
+Every action to make project Seata-go better is encouraged. On GitHub, every improvement for Seata-go could be via a PR (short for pull request).
+
+* If you find a typo, try to fix it!
+* If you find a bug, try to fix it!
+* If you find some redundant codes, try to remove them!
+* If you find some test cases missing, try to add them!
+* If you could enhance a feature, please **DO NOT** hesitate!
+* If you find code implicit, try to add comments to make it clear!
+* If you find code ugly, try to refactor that!
+* If you can help to improve documents, it could not be better!
+* If you find document incorrect, just do it and fix that!
+* ...
+
+Actually it is impossible to list them completely. Just remember one principle:
+
+> WE ARE LOOKING FORWARD TO ANY PR FROM YOU.
+
+Since you are ready to improve Seata-go with a PR, we suggest you could take a look at the PR rules here.
+
+* [Workspace Preparation](#workspace-preparation)
+* [Branch Definition](#branch-definition)
+* [Commit Rules](#commit-rules)
+* [PR Description](#pr-description)
+
+### Workspace Preparation
+
+To put forward a PR, we assume you have registered a GitHub ID. Then you could finish the preparation in the following steps:
+
+1. **FORK** Seata-go to your repository. To make this work, you just need to click the button Fork in right-left of [seata/seata](https://github.com/seata/seata-go) main page. Then you will end up with your repository in `https://github.com/<your-username>/seata-go`, in which `your-username` is your GitHub username.
+
+1. **CLONE** your own repository to develop locally. Use `git clone git@github.com:<your-username>/seata-go.git` to clone repository to your local machine. Then you can create new branches to finish the change you wish to make.
+
+1. **Set Remote** upstream to be `git@github.com:seata/seata-go.git` using the following two commands:
+
+```bash
+git remote add upstream git@github.com:seata/seata-go.git
+git remote set-url --push upstream no-pushing
+```
+
+With this remote setting, you can check your git remote configuration like this:
+
+```shell
+$ git remote -v
+origin     git@github.com:<your-username>/seata-golang.git (fetch)
+origin     git@github.com:<your-username>/seata-golang.git (push)
+upstream   git@github.com:seata/seata-golang.git (fetch)
+upstream   no-pushing (push)
+```
+
+Adding this, we can easily synchronize local branches with upstream branches.
+
+### Branch Definition
+
+Right now we assume every contribution via pull request is for [branch develop](https://github.com/seata/seata-go/tree/develop) in Seata-go. Before contributing, be aware of branch definition would help a lot.
```

Comment:
Seata-go, at the beginning of the sentence, it is Seata-go. If not, pls use the lower case seata-go instead.

Reply:
pls change it at other places too.